octodns/provider/ns1.py

```diff
@@ -35,11 +38,62 @@ def __init__(self, id, api_key, *args, **kwargs):
         self._client = NSONE(apiKey=api_key)
 
     def _data_for_A(self, _type, record):
-        return {
+        # record meta (which would include geo information is only
+        # returned when getting a record's detail, not from zone detail
+        geo = {}
+        data = {
             'ttl': record['ttl'],
             'type': _type,
-            'values': record['short_answers'],
         }
+        values, codes = [], []
+        if 'answers' not in record:
+            values = record['short_answers']
+        for answer in record.get('answers', []):
+            meta = answer.get('meta', {})
+            if meta:
+                # country + state and country + province are allowed
+                # in that case though, supplying a state/province would
+                # be redundant since the country would supercede in when
+                # resolving the record.  it is syntactically valid, however.
+                country = meta.get('country', [])
+                us_state = meta.get('us_state', [])
+                ca_province = meta.get('ca_province', [])
+                for cntry in country:
+                    cn = transformations.cc_to_cn(cntry)
+                    con = transformations.cn_to_ctca2(cn)
+                    key = '{}-{}'.format(con, cntry)
+                    if key not in geo:
```

Comment:
You could avoid having to do the `if key not in geo ... else ...` stuff in each block here by using `geo = defaultdict(list)` from `collections`.  then the `geo[key].extend(answer['answer']) should be enough.

Reply:
d'oh.  i forgot about defaultdict(list).